Tasks/PackerBuildV1/src/operations/packerBuild.ts

```diff
@@ -50,20 +50,32 @@ export async function run(packerHost: packerHost): Promise<any> {
 function setOutputVariables(packerHost: packerHost, outputs: Map<string, string>): void {
     var taskParameters = packerHost.getTaskParameters();
     var imageUri;
+    var imageId;
     var managedImageName;
+      
+    if (!utils.IsNullOrEmpty(taskParameters.imageId) && !(taskParameters.templateType == constants.TemplateTypeBuiltin && !taskParameters.isManagedImage)) {
+        imageId = tryGetValueFromOutputs(constants.PackerLogTokenManagedImageId, outputs);
+
+        if (!utils.IsNullOrEmpty(imageId)) {
+            tl.debug("Setting image Id variable which contains the managed image Id to: " + imageId);
+            tl.setVariable(taskParameters.imageId, imageId);
+        } else {
+            throw tl.loc("ImageIDOutputVariableNotFound");
```

Comment:
I think in that case it wouldn't get to the throw because it would fail the managed instance check in line 56.  I don't think changing line 63 to a warning will change anything.  @hiyadav  you have any thoughts?
```
tl.warning(tl.loc("ImageIDOutputVariableNotFound"));
```

Reply:
I thought the intention is to throw if we are not able to set the output variable with conditions meeting for it ot be set. In the scenario mentioned by @bishal-pdMSFT  error would be thrown, which is acceptable behaviour to me as the user has explicitly typed in the output variable name for ImageId.